R/split_moment_matching.R

```diff
@@ -0,0 +1,127 @@
+#' Split moment matching for efficient approximate leave-one-out cross-validation (LOO)
+#'
+#' A function that computes the split moment matching importance sampling loo.
+#' Takes in the moment matching total transformation, transforms only half
+#' of the draws, and computes a single elpd using multiple importance sampling.
+#'
+#'
+#'
+#'
+#' @param x A fitted model object.
+#' @param upars A matrix containing the model parameters in unconstrained space
+#' where they can have any real value.
+#' @param cov Logical; Indicate whether to match the covariance matrix of the
+#' samples or not. If \code{FALSE}, only the mean and marginal variances
+#' are matched.
+#' @param total_shift A vector representing the total shift made by the moment
+#' matching algorithm.
+#' @param total_scaling A vector representing the total scaling of marginal
+#' variance made by the moment matching algorithm.
+#' @param total_mapping A vector representing the total covariance
+#' transformation made by the moment matching algorithm.
+#' @param i Observation index.
+#' @param log_prob_upars A function that takes arguments \code{x} and
+#'   \code{upars} and returns a matrix of log-posterior density values of the
+#'   unconstrained posterior draws passed via \code{upars}.
+#' @param log_lik_i_upars A function that takes arguments \code{x}, \code{upars},
+#'   and \code{i} and returns a vector of log-likeliood draws of the \code{i}th
+#'   observation based on the unconstrained posterior draws passed via
+#'   \code{upars}.
+#' @param r_effi MCMC relative effective sample size of the \code{i}'th
+#' log likelihood draws.
+#' @template cores
+#' @template is_method
+#' @param ... Further arguments passed to the custom functions documented above.
+#'
+#' @return A list containing the updated log-importance weights and
+#' log-likelihood values. Also returns the updated MCMC effective sample size
+#' and the integrand-specific log-importance weights.
+#'
+#'
+#' @seealso [loo()], [mmloo()]
+#' @template moment-matching-references
+#'
+#'
+split_mmloo <- function(x, upars, cov, total_shift, total_scaling,
+                     total_mapping, i, log_prob_upars,
+                     log_lik_i_upars, r_effi, cores,
```

Comment:
Do you mind changing the name of the argument from `r_effi` to `r_eff_i` for consistency with other names?

Reply:
Thanks! This is now changed